### ISSUE TITLE
运行印章识别pgnet算法报错

### ISSUE THREAD

**Issue by LeFuGang** [PaddlePaddle/PaddleOCR#9064]
请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem

- 系统环境/System Environment：
- 版本号/Version：Paddle：  PaddleOCR： 问题相关组件/Related components：
- 运行指令/Command Code：
- 完整报错/Complete Error Message：
我下载了paddlecor开源大礼包中的模型，想运行印章识别的模型，我环境如下：
我在镜像paddlecloud/paddleocr，版本2.6-gpu-cuda11.2-cudnn8-latest中运行的，
![image](https://user-images.githubusercontent.com/28383188/218634841-2adfd0e1-e96b-42a3-b65d-f0cd5b532a00.png)
这是官方开源的模型和配置文件，
这是我运行的脚本：
python tools/infer_e2e.py -c configs/seal_models/PGNet/config.yml -o Global.pretrained_model=configs/seal_models/PGNet/best_accuracy Global.infer_img=data/1ttt.jpg 

但是报错了，其他模型都能运行正常，我很奇怪，是不是版本问题尼？
![image](https://user-images.githubusercontent.com/28383188/218635094-7ea493ba-9dd1-4460-add2-32e7354daa8c.png)
请专家帮忙看下，谢谢了

**Comment by LeFuGang** [comment]
> 请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem
> 
> * 系统环境/System Environment：
> * 版本号/Version：Paddle：  PaddleOCR： 问题相关组件/Related components：
> * 运行指令/Command Code：
> * 完整报错/Complete Error Message：
>   我下载了paddlecor开源大礼包中的模型，想运行印章识别的模型，我环境如下：
>   我在镜像paddlecloud/paddleocr，版本2.6-gpu-cuda11.2-cudnn8-latest中运行的，
>   ![image](https://user-images.githubusercontent.com/28383188/218634841-2adfd0e1-e96b-42a3-b65d-f0cd5b532a00.png)
>   这是官方开源的模型和配置文件，
>   这是我运行的脚本：
>   python tools/infer_e2e.py -c configs/seal_models/PGNet/config.yml -o Global.pretrained_model=configs/seal_models/PGNet/best_accuracy Global.infer_img=data/1ttt.jpg
> 
> 但是报错了，其他模型都能运行正常，我很奇怪，是不是版本问题尼？ ![image](https://user-images.githubusercontent.com/28383188/218635094-7ea493ba-9dd1-4460-add2-32e7354daa8c.png) 请专家帮忙看下，谢谢了

有大佬能反馈下问题吗，是哪里的问题呀，版本吗

**Comment by andyjpaddle** [comment]
加载的模型和配置文件没对应上，发布的模型是中文的，需要用中文字典

**Comment by LeFuGang** [comment]
> 模型和配置文件没对

Global:
  use_gpu: true
  epoch_num: 600
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/pgnet_r50_vd_seal_run/
  save_epoch_step: 300
  eval_batch_step:
  - 0
  - 400
  cal_metric_during_train: false
  pretrained_model: ./pretrain_models/train_step1/best_accuracy
  checkpoints: null
  save_inference_dir: null
  use_visualdl: false
  infer_img: null
  valid_set: totaltext
  save_res_path: ./output/pgnet_r50_vd_totaltext/predicts_pgnet.txt
  character_dict_path: ppocr/utils/ppocr_keys_v1.txt
  character_type: EN
  max_text_length: 50
  max_text_nums: 10
  tcl_len: 64
  load_static_weights: false
Architecture:
  model_type: e2e
  algorithm: PGNet
  Transform: null
  Backbone:
    name: ResNet
    layers: 50
  Neck:
    name: PGFPN
  Head:
    name: PGHead
Loss:
  name: PGLoss
  tcl_bs: 32
  max_text_length: 50
  max_text_nums: 10
  pad_num: 6623
Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.001
    warmup_epoch: 50
  regularizer:
    name: L2
    factor: 5.0e-05
PostProcess:
  name: PGPostProcess
  score_thresh: 0.5
  mode: fast
Metric:
  name: E2EMetric
  mode: A
  gt_mat_dir: ./train_data/total_text/gt
  character_dict_path: ppocr/utils/ppocr_keys_v1.txt
  main_indicator: f_score_e2e
Train:
  dataset:
    name: PGDataSet
    data_dir: ./train_data/
    label_file_list:
    - ./train_data/label_2_rotate_transform_final.txt
    - ./train_data/label_3_rotate_transform_final.txt
    ratio_list:
    - 1.0
    - 1.0
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - E2ELabelEncodeTrain: null
    - PGProcessTrain:
        batch_size: 5
        min_crop_size: 24
        min_text_size: 4
        max_text_size: 512
    - KeepKeys:
        keep_keys:
        - images
        - tcl_maps
        - tcl_label_maps
        - border_maps
        - direction_maps
        - training_masks
        - label_list
        - pos_list
        - pos_mask
  loader:
    shuffle: true
    drop_last: true
    batch_size_per_card: 5
    num_workers: 16
Eval:
  dataset:
    name: PGDataSet
    data_dir: ./train_data/
    label_file_list:
    - ./train_data/label_test.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - E2ELabelEncodeTest:
        character_dict_path: ppocr/utils/ppocr_keys_v1.txt
    - DetResizeForTest:
        image_shape:
        - 512
        - 512
        padding: true
    - ToCHWImage: null
    - KeepKeys:
        keep_keys:
        - image
        - shape
        - polys
        - texts
        - ignore_tags
        - img_id
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 1
    num_workers: 2
profiler_options: null

这是官方给得配置文件，需要改哪里尼

**Comment by andyjpaddle** [comment]
https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/ppocr/modeling/heads/e2e_pg_head.py#L71  这里默认是英文字典，需要改下配置文件中的Head字段，加一个参数：`character_dict_path: ppocr/utils/ppocr_keys_v1.txt`

**Comment by LeFuGang** [comment]
> character_dict_path: ppocr/utils/ppocr_keys_v1.txt

![image](https://user-images.githubusercontent.com/28383188/219521434-3014a78f-1b88-4b5b-be08-17b4ea030fbf.png)
![image](https://user-images.githubusercontent.com/28383188/219521464-44c53bc8-ad17-45ef-a558-8d375cf4ae8b.png)

我加了这个参数，而且我把PGHead都改了，还是报一样的错误，人都麻了

**Comment by LDOUBLEV** [comment]
没有做归一化，- ToCHWImage:之前加上
     
 - NormalizeImage:
          scale: 1./255.
          mean: [ 0.485, 0.456, 0.406 ]
          std: [ 0.229, 0.224, 0.225 ]
          order: 'hwc'

**Comment by LeFuGang** [comment]
> character_dict_path: ppocr/utils/ppocr_keys_v1.txt

![image](https://user-images.githubusercontent.com/28383188/219612825-6226c755-8b08-416f-be07-08b5bf97b0e1.png)
报错，我就添加了这个参数，不报错了，但是结果有问题，
![image](https://user-images.githubusercontent.com/28383188/219612928-00a40a6b-e15b-4158-9d35-ba4b36b15351.png)
![image](https://user-images.githubusercontent.com/28383188/219613167-3942e7f4-ece7-4ed3-b536-3dbb9726cc94.png)

config.yml文件我添加了你给的两个参数
![image](https://user-images.githubusercontent.com/28383188/219613330-0db7396c-02ff-4ee4-b1f8-4fbf2887b79d.png)
![image](https://user-images.githubusercontent.com/28383188/219613434-4fdecc54-d1d1-41c1-ac83-1fe3146830de.png)

我想问下为什么结果这个样子呀，跟官方的结果不一样尼

下面是官方推理的（https://aistudio.baidu.com/aistudio/projectdetail/4983788?channelType=0&channel=0）
![image](https://user-images.githubusercontent.com/28383188/219613816-e563c776-24db-4695-87d5-f6e8fb8931cd.png)

**Comment by LeFuGang** [comment]
> 没有做归一化，- ToCHWImage:之前加上
> 
> * NormalizeImage:
>   scale: 1./255.
>   mean: [ 0.485, 0.456, 0.406 ]
>   std: [ 0.229, 0.224, 0.225 ]
>   order: 'hwc'

大佬能回答下吗，到底哪里的问题尼

**Comment by wangxinzhe0617** [comment]
您好大佬，能提供一下seal_models/PGNet/best_accuracy推理模型不，现在好像找不到了。万分感谢。

**Comment by LeFuGang** [comment]
> 没有做归一化，- ToCHWImage:之前加上
> 
> * NormalizeImage:
>   scale: 1./255.
>   mean: [ 0.485, 0.456, 0.406 ]
>   std: [ 0.229, 0.224, 0.225 ]
>   order: 'hwc'

Global:
  use_gpu: True
  epoch_num: 600
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/pgnet_r50_vd_totaltext/
  save_epoch_step: 10
  # evaluation is run every 0 iterationss after the 1000th iteration
  eval_batch_step: [ 0, 1000 ]
  cal_metric_during_train: False
  pretrained_model:
  checkpoints:
  save_inference_dir:
  use_visualdl: False
  infer_img:
  infer_visual_type: CN # two mode: EN is for english datasets, CN is for chinese datasets
  valid_set: totaltext # two mode: totaltext valid curved words, partvgg valid non-curved words
  save_res_path: ./output/pgnet_r50_vd_totaltext/predicts_pgnet.txt
  character_dict_path: ppocr/utils/ppocr_keys_v1.txt
  character_type: CN
  max_text_length: 50 # the max length in seq
  max_text_nums: 30 # the max seq nums in a pic
  tcl_len: 64

Architecture:
  model_type: e2e
  algorithm: PGNet
  Transform:
  Backbone:
    name: ResNet
    layers: 50
  Neck:
    name: PGFPN
  Head:
    name: PGHead
    character_dict_path: ppocr/utils/ppocr_keys_v1.txt # the same as Global:character_dict_path

Loss:
  name: PGLoss
  tcl_bs: 64
  max_text_length: 50 # the same as Global: max_text_length
  max_text_nums: 30 # the same as Global：max_text_nums
  pad_num: 36 # the length of dict for pad

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.001
    warmup_epoch: 50
  regularizer:
    name: 'L2'
    factor: 0.0001

PostProcess:
  name: PGPostProcess
  score_thresh: 0.5
  mode: fast   # fast or slow two ways
  point_gather_mode: align # same as PGProcessTrain: point_gather_mode

Metric:
  name: E2EMetric
  mode: A   # two ways for eval, A: label from txt,  B: label from gt_mat
  gt_mat_dir:  ./train_data/total_text/gt  # the dir of gt_mat
  character_dict_path: ppocr/utils/ppocr_keys_v1.txt
  main_indicator: f_score_e2e

Train:
  dataset:
    name: PGDataSet
    data_dir: ./train_data/total_text/train
    label_file_list: [./train_data/total_text/train/train.txt]
    ratio_list: [1.0]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - E2ELabelEncodeTrain:
      - PGProcessTrain:
          batch_size: 14  # same as loader: batch_size_per_card
          use_resize: True
          use_random_crop: False
          min_crop_size: 24
          min_text_size: 4
          max_text_size: 512
          point_gather_mode: align # two mode: align and none, align mode is better than none mode
      - KeepKeys:
          keep_keys: [ 'images', 'tcl_maps', 'tcl_label_maps', 'border_maps','direction_maps', 'training_masks', 'label_list', 'pos_list', 'pos_mask' ] # dataloader will return list in this order
  loader:
    shuffle: True
    drop_last: True
    batch_size_per_card: 14
    num_workers: 16

Eval:
  dataset:
    name: PGDataSet
    data_dir: ./train_data/total_text/test
    label_file_list: [./train_data/total_text/test/test.txt]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - E2ELabelEncodeTest:
          character_dict_path: ppocr/utils/ppocr_keys_v1.txt
      - E2EResizeForTest:
          max_side_len: 768
          image_shape:
          - 512
          - 512
          padding: true
      - NormalizeImage:
          scale: 1./255.
          mean: [ 0.485, 0.456, 0.406 ]
          std: [ 0.229, 0.224, 0.225 ]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: [ 'image', 'shape', 'polys', 'texts', 'ignore_tags', 'img_id']
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 1 # must be 1
    num_workers: 2
这是我的config文件，但是我的结果识别有点不一样，最下面的“专用章”没有识别到：
![image](https://user-images.githubusercontent.com/28383188/220240012-194db6f8-e68b-48ff-9129-106f1aed5856.png)

请问下这是什么问题尼

**Comment by LeFuGang** [comment]
> 您好大佬，能提供一下seal_models/PGNet/best_accuracy推理模型不，现在好像找不到了。万分感谢。

链接: https://pan.baidu.com/s/1048jo1Gll_kGxpDM8rqSSw?pwd=1234 提取码: 1234 复制这段内容后打开百度网盘手机App，操作更方便哦@youngdream

**Comment by wangxinzhe0617** [comment]
> > 您好大佬，能提供一下seal_models/PGNet/best_accuracy推理模型不，现在好像找不到了。万分感谢。
> 
> 链接: https://pan.baidu.com/s/1048jo1Gll_kGxpDM8rqSSw?pwd=1234 提取码: 1234 复制这段内容后打开百度网盘手机App，操作更方便哦@YoungDream

谢谢大佬！^^

**Comment by sadays22** [comment]
> > 没有做归一化，- ToCHWImage:之前加上
> > 
> > * NormalizeImage:
> >   scale: 1./255.
> >   mean: [ 0.485, 0.456, 0.406 ]
> >   std: [ 0.229, 0.224, 0.225 ]
> >   order: 'hwc'
> 
> Global: use_gpu: True epoch_num: 600 log_smooth_window: 20 print_batch_step: 10 save_model_dir: ./output/pgnet_r50_vd_totaltext/ save_epoch_step: 10
> 
> # evaluation is run every 0 iterationss after the 1000th iteration
> eval_batch_step: [ 0, 1000 ] cal_metric_during_train: False pretrained_model: checkpoints: save_inference_dir: use_visualdl: False infer_img: infer_visual_type: CN # two mode: EN is for english datasets, CN is for chinese datasets valid_set: totaltext # two mode: totaltext valid curved words, partvgg valid non-curved words save_res_path: ./output/pgnet_r50_vd_totaltext/predicts_pgnet.txt character_dict_path: ppocr/utils/ppocr_keys_v1.txt character_type: CN max_text_length: 50 # the max length in seq max_text_nums: 30 # the max seq nums in a pic tcl_len: 64
> 
> Architecture: model_type: e2e algorithm: PGNet Transform: Backbone: name: ResNet layers: 50 Neck: name: PGFPN Head: name: PGHead character_dict_path: ppocr/utils/ppocr_keys_v1.txt # the same as Global:character_dict_path
> 
> Loss: name: PGLoss tcl_bs: 64 max_text_length: 50 # the same as Global: max_text_length max_text_nums: 30 # the same as Global：max_text_nums pad_num: 36 # the length of dict for pad
> 
> Optimizer: name: Adam beta1: 0.9 beta2: 0.999 lr: name: Cosine learning_rate: 0.001 warmup_epoch: 50 regularizer: name: 'L2' factor: 0.0001
> 
> PostProcess: name: PGPostProcess score_thresh: 0.5 mode: fast # fast or slow two ways point_gather_mode: align # same as PGProcessTrain: point_gather_mode
> 
> Metric: name: E2EMetric mode: A # two ways for eval, A: label from txt, B: label from gt_mat gt_mat_dir: ./train_data/total_text/gt # the dir of gt_mat character_dict_path: ppocr/utils/ppocr_keys_v1.txt main_indicator: f_score_e2e
> 
> Train: dataset: name: PGDataSet data_dir: ./train_data/total_text/train label_file_list: [./train_data/total_text/train/train.txt] ratio_list: [1.0] transforms: - DecodeImage: # load image img_mode: BGR channel_first: False - E2ELabelEncodeTrain: - PGProcessTrain: batch_size: 14 # same as loader: batch_size_per_card use_resize: True use_random_crop: False min_crop_size: 24 min_text_size: 4 max_text_size: 512 point_gather_mode: align # two mode: align and none, align mode is better than none mode - KeepKeys: keep_keys: [ 'images', 'tcl_maps', 'tcl_label_maps', 'border_maps','direction_maps', 'training_masks', 'label_list', 'pos_list', 'pos_mask' ] # dataloader will return list in this order loader: shuffle: True drop_last: True batch_size_per_card: 14 num_workers: 16
> 
> Eval: dataset: name: PGDataSet data_dir: ./train_data/total_text/test label_file_list: [./train_data/total_text/test/test.txt] transforms: - DecodeImage: # load image img_mode: BGR channel_first: False - E2ELabelEncodeTest: character_dict_path: ppocr/utils/ppocr_keys_v1.txt - E2EResizeForTest: max_side_len: 768 image_shape: - 512 - 512 padding: true - NormalizeImage: scale: 1./255. mean: [ 0.485, 0.456, 0.406 ] std: [ 0.229, 0.224, 0.225 ] order: 'hwc' - ToCHWImage: - KeepKeys: keep_keys: [ 'image', 'shape', 'polys', 'texts', 'ignore_tags', 'img_id'] loader: shuffle: False drop_last: False batch_size_per_card: 1 # must be 1 num_workers: 2 这是我的config文件，但是我的结果识别有点不一样，最下面的“专用章”没有识别到： ![image](https://user-images.githubusercontent.com/28383188/220240012-194db6f8-e68b-48ff-9129-106f1aed5856.png)
> 
> 请问下这是什么问题尼

这个还能识别出来中文么
